enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,217 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@EmilienM"
+  - "@mdbooth"
+reviewers:
+  - "@mandre"
+  - "@pierreprinetti"
+approvers:
+  - "@mandre"
+  - "@pierreprinetti"
+creation-date: 2020-11-04
+last-updated: 2020-11-04
+status: implementable
+---
+
+# OpenStack supported versions
+
+- [OpenStack supported versions](#openstack-supported-versions)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Design Details](#design-details)
+    - [User Stories](#user-stories)
+      - [Check the OpenStack version](#check-the-openstack-version)
+      - [Ignore the version check](#ignore-the-version-check)
+        - [Example Usage](#example-usage)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Dev Preview -> Tech Preview](#dev-preview---tech-preview)
+      - [Tech Preview -> GA](#tech-preview---ga)
+    - [Infrastructure Needs](#infrastructure-needs)
+    - [Drawbacks](#drawbacks)
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+OpenShift can only be installed on supported versions of OpenStack.
+While the [Support Matrix](https://access.redhat.com/articles/4679401) is
+well documented, the OpenShift installer doesn't check which version of
+OpenStack is running before attempting an operation (e.g. installation). 
+Running OpenShift over an unsupported version of OpenStack is problematic
+and can lead to unstable infrastructure. To avoid that, it is proposed
+to programmatically verify the version of OpenStack is supported
+otherwise stop and fail.
+
+## Motivation
+
+### Goals
+
+- Disallow any kind of OpenShift operation done through the installer to run
+  against a cloud where the OpenStack version is not supported.
```

Comment:
From a user perspective, "any kind of OpenShift operation" includes applying a deployment or even stopping a pod. I suspect that here we actually only refer to operations that involve the underlying OpenStack, and even more precisely only "install" and "upgrade". If it is the case, shall we list those explicitly?

I am not mentioning "downgrade", because I expect that operation to be carried out in emergency cases, where we don't want to introduce additional friction.

Reply:
ack, will change the wording.